nomos-services/api/src/http/da.rs

```diff
@@ -269,3 +280,89 @@ where
 
     receiver.await?
 }
+
+pub async fn block_peer<B>(handle: &OverwatchHandle, peer_id: PeerId) -> Result<bool, DynError>
+where
+    B: NetworkBackend + 'static + Send,
+    B::Message: PeerMessagesBuilder,
+{
+    let relay = handle.relay::<NetworkService<B>>().connect().await?;
+    let (sender, receiver) = oneshot::channel();
+    let message = B::Message::create_block_message(peer_id, sender);
+    relay
+        .send(DaNetworkMsg::Process(message))
+        .await
+        .map_err(|(e, _)| e)?;
+    Ok(receiver.await?)
+}
+
+pub async fn unblock_peer<B>(handle: &OverwatchHandle, peer_id: PeerId) -> Result<bool, DynError>
+where
+    B: NetworkBackend + 'static + Send,
+    B::Message: PeerMessagesBuilder,
+{
+    let relay = handle.relay::<NetworkService<B>>().connect().await?;
+    let (sender, receiver) = oneshot::channel();
+    let message = B::Message::create_unblock_message(peer_id, sender);
+    relay
+        .send(DaNetworkMsg::Process(message))
+        .await
+        .map_err(|(e, _)| e)?;
+    Ok(receiver.await?)
+}
+
+pub async fn blacklisted_peers<B>(handle: &OverwatchHandle) -> Result<Vec<PeerId>, DynError>
+where
+    B: NetworkBackend + 'static + Send,
+    B::Message: PeerMessagesBuilder,
+{
+    let relay = handle.relay::<NetworkService<B>>().connect().await?;
+    let (sender, receiver) = oneshot::channel();
+    let message = B::Message::create_blacklisted_message(sender);
+    relay
+        .send(DaNetworkMsg::Process(message))
+        .await
+        .map_err(|(e, _)| e)?;
+    Ok(receiver.await?)
+}
+
+// Builder for generating messages for peers (validator and executor).
+pub trait PeerMessagesBuilder {
+    type MessageType;
```

Comment:
What's the goal of this associated type here? All methods return `Self`, and this type is `Self` for all impls, so I am not sure I understand its purpose?

Reply:
There is a duplication of messages in Validator and Executor, so they have "different messages. I talked with the guys and consensus was to keep those separated. So the message that needs to be created is either `ExecutorDaNetworkMessage` or `DaNetworkMessage` (validator one), so there are correspond implementations for those, which are the same as they both create Self::PeerRequest, which is defined in `nomos/nomos-services/data-availability/network/src/backends/libp2p/common`.
So I added a simple builder trait to build appropriate message based on the Message type.

Reply:
I edited my message, I see what you mean now, yeah the type does not have to be included at all. Good catch!

Reply:
Fixed. thank you!

Reply:
The way I would do this, is to rename the associated type to `Message` and for the methods to return `Self::Message`. For a builder to return `Self` means, at least in theory, that the builder is returning itself. I guess what we want from the builder are messages, right?

Reply:
Then you can add different trait bounds in different places where such message type is needed, without having to constraint the builder itself.

Reply:
Maybe I am missing something, feel free to show the example, pls. The way I just fixed eliminates the associated type completely, and fits the calling in handlers perfectly. I am not sure if we need to support any more use cases for this, tbh.

Reply:
What I mean is that it feels a bit unconventional for a builder to return `Self`, rather than `Message`. I think it would work just fine with a `Message` associated type and the methods returning `Self::Message` instead. But maybe we can wait for someone else to provide their input. I am not saying your solution does not work, as it clearly seems it does, but I am saying that the semantics of what I see here are perhaps not 100% "to the book".

Reply:
I see what you mean now. Yes, I think that for builders it is common to return self and have a `.build()` method in the end. But I have also seen across Rust places where builder type is actually a Message type, which is seen from the implementation and I feel like it is contributing to simplicity. That is why I have chosen the pattern, as we don't have a parametrized builder. 

But I don't have any strong feelings about it, happy to change if it's a bad idiom.

@bacv, @danielSanchezQ  let us know what do you guys think.

Reply:
Sorry I just noticed that the trait is not implemented for the builder, but for the actual message type already, for instance `DaMessage`. I think I was confused at first, but now I see it again, there is definitely not need for an associated type. Sorry for this misunderstanding, I think the current versions looks ok as it is 👍

Reply:
But I'll still let someone also approve the PR as a whole 😊

Reply:
Thinking about this more, maybe this is more of a factory then builder, so I will change the name :)

Reply:
> What I mean is that it feels a bit unconventional for a builder to return `Self`, rather than `Message`. I think it would work just fine with a `Message` associated type and the methods returning `Self::Message` instead. But maybe we can wait for someone else to provide their input. I am not saying your solution does not work, as it clearly seems it does, but I am saying that the semantics of what I see here are perhaps not 100% "to the book".

Ok, I can see the convo now after the review.
Check my comment here even if it is not relevant anymore: https://github.com/logos-co/nomos/pull/1131#discussion_r1995121942

Reply:
@ntn-x2 Are you ok to merge this now?

Reply:
Sure, go ahead!